test/compare/default/class_declaration-in.js

```diff
@@ -33,3 +33,15 @@ return this.properties;
 }
 
 }
+
+// Multi line declaration
+class
+Foo
+extends
+Bar
+{
+}
+// The value to be extended can be produced by an arbitrary expression.
+class Foo extends BarNamespace.Bar{}
+class Foo extends BarNamespace [ 'Bar' ] . Bar [ "Bar" ] ( 'bar' ) . Bar ( "bar", "bar" ){}
```

Comment:
I'm going to merge your patch but I think I'm going to ignore this edge case (ignore the fact that input might contain weird white spaces in between the properties).. since it's complex and not really part of the `ClassDeclaration` (should be handled by `MemberExpression` and `CallExpression` instead)